MIPs/mip-mc-handshake.md

```diff
@@ -0,0 +1,255 @@
+
+## Rationale
+By incorporating CAIP-25 and EIP-3085 standards, this proposal aims to provide a comprehensive and standards-friendly way for dApp developers to request wallet permissions in a multichain context. The `provider_authorize` method will also enable an intuitive connection experience for users who wish to interact with dApps across multiple chains using MetaMask. Developers can choose to either request permissions upfront or to request them progressively according to their needs.
+
+## Test Cases
+Test cases should cover:
+- Successful provider_authorize requests with expected permissions returned in the response.
+- Error handling when an unsupported scope object or permission is requested
+- Error handling when the user denies consent for requested methods
+- Error handling when the user denies all requested or any required scope objects
+  
+## Implementation
+Implementing this proposal involves a significant modification to MetaMask's JSON-RPC request-handling code to support the new `provider_authorize` method, adjustments to the permissioning system, and updates to the UI components for user consent and management of dapp permissions.
+
+## Developer Adoption Considerations
+
+[Explain any considerations that developers should take into account when adopting this proposal. For example, how will it affect compatibility, and what changes may need to be made to accommodate the proposal?]
+
+Backward compatibility will be maintained through the existing Ethereum Provider API. However, the Multichain API WILL NOT be available through the existing Ethereum Provider API. Because many developers rely on third-party libraries to connect their dapps with wallets, a some mapping logic that allows them to keep their "single chain" code as-is while actually hitting the multichain API may facilitate more rapid adoption. 
+
+Developers should expect most new improvements to the Wallet API to be delivered only though the multichain API, NOT the Ethereum Provider API.
+
+## User Experience Considerations
+
+Calling the `provider_authorize` method with `optionalScopes` that include any eip155 prefixed scopes will trigger the following sequence:
+
+```mermaid
+sequenceDiagram
+    app->>+MetaMask: provider_authorize Request
+    actor User
+    MetaMask->>User: Network Permissions
+    activate User
+    MetaMask->>User: Acccount Permissions
+    User-->>MetaMask: Confirm
+    deactivate User
+    MetaMask-->>-app: provider_authorize Response
+```
+
+## Privacy Considerations
+This proposal raises important privacy considerations, including the need to avoid data leaking and the challenge of obtaining genuine user consent. It underscores the importance of preserving user anonymity and the sensitivites involved in determining authorizations. Identifying and mitigating these issues is crucial for protecting user privacy during multichain interactions, prompting a careful evaluation of how best to balance functionality with privacy concerns.
```

Comment:
Echoing what @bumblefudge above, it would be prudent to mirror CAIP-25's error handling to avoid things like fingerprinting risk through authorization denials/errors.